File: python/S01-elemental-actions/E03-binary-to-decimal.py
def main():
	print("\n\033[35m[========= BINARY TO DECIMAL =========]\033[0m\n")

	binary: str = input("Enter a binary number: ")

	first: int = int(binary[3])
	second: int = int(binary[2])
	third: int = int(binary[1])
	fourth: int = int(binary[0])

	decimal: int = (first * 2 ** 0) + (second * 2 ** 1) + (third * 2 ** 2) +  (fourth * 2 ** 3)

	print(f"\nThe result of inverting the number is: {decimal}")


if __name__ == "__main__":
	main()

File: c++/S01-elemental-actions/E03-binary-to-decimal.cpp
#include <iostream>

int main(int argc, char *argv[]) {
	std::cout << "\n\e[0;35m[========= BINARY TO DECIMAL =========]\e[0m\n\n";

	int binaryNumber, decimalNumber;

	std::cout << "Enter a 4 digit binary number: ";
	std::cin >> binaryNumber;

	int firstDigit, secondDigit, thirdDigit, fourthDigit;

	firstDigit = binaryNumber / 1000;
	secondDigit = (binaryNumber % 1000) / 100;
	thirdDigit = (binaryNumber % 100) / 10;
	fourthDigit = binaryNumber % 10;

	decimalNumber = (firstDigit * 8) + (secondDigit * 4) + (thirdDigit * 2) + fourthDigit;

	std::cout << "The decimal value is: " << decimalNumber << '\n';

	return 0;
}
